src/node/miner.cpp

```diff
@@ -120,6 +120,12 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     pblocktemplate->vTxFees.push_back(-1); // updated at end
     pblocktemplate->vTxSigOpsCost.push_back(-1); // updated at end
 
+    // Use dummy coinbase output script if none is provided
+    CScript coinbase_output_script{CScript() << OP_TRUE};
+    if (m_options.coinbase_output_script) {
+        coinbase_output_script = m_options.coinbase_output_script.value();
```

Comment:
In commit "Drop script_pub_key arg from createNewBlock" (42f880aeb080070a42229d0bccd517100a3fe7fb)

Would suggest using '*' instead of `.value()` since this code isn't intending to trigger the `std::bad_optional_access` exception, and to make it more obvious a dereference is happening.

Reply:
I changed `coinbase_output_script` to no longer be `std::optional`.